crates/ingress-http/src/handler/tracing.rs

```diff
@@ -11,59 +11,33 @@
 use super::ConnectInfo;
 
 use http::Request;
-use opentelemetry::trace::{SpanContext, TraceContextExt};
+use opentelemetry::global::ObjectSafeSpan;
+use opentelemetry::trace::SpanContext;
+use restate_tracing_instrumentation as instrumentation;
 use restate_types::identifiers::InvocationId;
-use restate_types::invocation::{InvocationTarget, SpanRelation};
-use tracing::{info_span, Span};
-use tracing_opentelemetry::OpenTelemetrySpanExt;
+use restate_types::invocation::InvocationTarget;
 
 pub(crate) fn prepare_tracing_span<B>(
     invocation_id: &InvocationId,
     invocation_target: &InvocationTarget,
     req: &Request<B>,
-) -> (Span, SpanContext) {
+) -> SpanContext {
     let connect_info: &ConnectInfo = req
         .extensions()
         .get()
         .expect("Should have been injected by the previous layer");
     let (client_addr, client_port) = (connect_info.address(), connect_info.port());
 
-    // Create the ingress span and attach it to the next async block.
-    // This span is committed once the async block terminates, recording the execution time of the invocation.
-    // Another span is created later by the ServiceInvocationFactory, for the ServiceInvocation itself,
-    // which is used by the Restate services to correctly link to a single parent span
-    // to commit intermediate results of the processing.
-    let ingress_span = info_span!(
-        "ingress_invoke",
-        otel.name = format!("ingress_invoke {}", invocation_target),
-        rpc.system = "restate",
-        rpc.service = %invocation_target.service_name(),
-        rpc.method = %invocation_target.handler_name(),
-        restate.invocation.id = %invocation_id,
-        restate.invocation.target = %invocation_target,
-        client.socket.address = %client_addr,
-        client.socket.port = %client_port,
+    let span = instrumentation::info_invocation_span!(
+        relation = SpanRelation::None,
+        prefix = "ingress",
+        id = invocation_id,
+        target = invocation_target,
+        tags = (
+            client.socket.address = client_addr.to_string(),
+            client.socket.port = client_port as i64
+        )
     );
 
-    // Extract tracing context if any
-    let tracing_context: &opentelemetry::Context = req
-        .extensions()
-        .get()
-        .expect("Should have been injected by the previous layer");
-
-    // Attach this ingress_span to the parent parsed from the headers, if any.
```

Comment:
i think we used to support inbound traceparent header and include it with the ingress span. is anything like that still present? do we want it to be?

Reply:
Good catch! seems this was lost during refactoring. I assume this should become the parent for the service span only, no ?

Reply:
i dont know if it should be the parent for the logical invocation span. perhaps it should be the parent for the ingress span, and we already have the follows_from -> the logical invocation span. ideally the same would apply for ingress spans related to scheduling delayed invocations, resolving awakeables etc.

Reply:
I think you are on the same page, just using different terminology here. (Service Span == Ingress Span, as in the ingress span that we emit to OTEL directly), no?

Reply:
we have 2 kind of spans only:
1) runtime spans, these are spans that you create with `tracing::span!` macro and its variants.
2) services spans, these are (otel) spans that you create with the new macro `invocation_span!` and its variants. Those spans you create normally associated with an invocation SpanContext (ServiceInvocationSpanContext) and you only create this span in some points of interest to carefully craft the user services view. Any invocation span links automatically to the runtime span that creates it, hence we have direct links from user spans to runtime spans.

Ingress is the same. we create a runtime ingress span, and a service ingress span (that links to the runtime ingress span) the service ingress span becomes the parent span of the Invocation object (ServiceInvocationSpanContext). Hence later on when you create any other invocation span, it links to that full user trace.

Reply:
ok, i think it makes sense for the inbound trace header to be the parent of the service ingress span in that case